src/controllers/bus.controller.ts

```diff
@@ -1,52 +1,185 @@
-import {BusesModel} from '../models/bus.model';
-import express, {Request,Response} from 'express';
-import {IBus} from '../interface/bus.interface';
-
-
-const router=express.Router();
-
-router.post('/createBus/', async(req:Request,res:Response)=>{
-    const bus=new BusesModel({
-        line_number:req.body.line_number,
-        bus_color:req.body.bus_color,
-        model:req.body.model,
-        speed:req.body.speed,
-        List_station:req.body.List_station
-    })
-    try{
-        const newBus= await bus.save();
-        res.status(201).json(newBus);
-    }catch(err){
-        res.status(400).json({message:err.message}) 
-    }
-});
-
-router.get('/readBus/:line_number',async(req:Request,res:Response)=>{
-    try{
-        const readBus:IBus=await BusesModel.findOne({line_number: req.params.line_number});
-        res.send(readBus)
-    }catch(err){
-        res.status(500).json({message:err.message})
-    }
-});
-
-router.get('/readBuses/',async(req:Request,res:Response)=>{
-    try{
-        const readBuses:IBus=await BusesModel.find();
-        res.send(readBuses)
-    }catch(err){
-        res.status(500).json({message:err.message})
+/* eslint-disable prefer-const */
+/* eslint-disable prefer-destructuring */
+/* eslint-disable import/no-unresolved */
+/* eslint-disable import/no-extraneous-dependencies */
+/* eslint-disable radix */
+/* eslint-disable no-multi-assign */
+/* eslint-disable no-shadow */
+/* eslint-disable camelcase */
+import { Request, Response } from 'express';
+import { BusesModel } from '../models/bus.model';
+import { IBus } from '../interface/bus.interface';
+
+export const postBus = async (req: Request, res: Response) => {
+  const line_num = await BusesModel.aggregate([
+    {
+      $project: {
+        line_number: '$line_number',
+      },
+    },
+  ]);
+  const num: number[] = [];
+  line_num.forEach((doc) => {
+    num.push(doc.line_number);
+  });
+
+  const bus = new BusesModel({
+    line_number: req.body.line_number,
+    bus_color: req.body.bus_color,
+    model: req.body.model,
+    speed: req.body.speed,
+    stationsList: req.body.stationsList,
+  });
+  const chck: any = req.body.line_number;
+  if (!num.includes(chck)) {
+    try {
+      const newBus = await bus.save();
+      return res.status(201).json(newBus);
+    } catch (err) {
+      return err;
     }
-})
-
-router.patch('/updateBus/:update',async(req:Request,res:Response)=>{
-    try{
-        const update_bus : IBus= await BusesModel.findByIdAndUpdate({line_number:49, model:1800});
-        res.send(update_bus)
-    }catch(err){
-        res.status(500).json({message:err.message})
+  } else {
+    return res.send('The line number exists');
+  }
+};
+
+export const getOne = async (req: Request, res: Response) => {
+  try {
+    const readBus: IBus = await BusesModel.findOne({
+      bus_color: req.params.bus_color,
+    });
+    return res.send(readBus);
+  } catch (err) {
+    return err;
+  }
+};
```

Comment:
This contains both the controller and the repository. the controller should not "talk" to the mongoose directly. 
Sseperate to anothe `repository` file.